docs/pages/project/contributing/contributing-docs.md

```diff
@@ -24,15 +24,101 @@ Meshery documentation is made of these components:
+- Fire up your WSL VM and install the ruby version manager (RVM): 
+  ```bash
+  sudo apt update
+  sudo apt install curl g++ gnupg gcc autoconf automake bison build-essential libc6-dev \
+    	libffi-dev libgdbm-dev libncurses5-dev libsqlite3-dev libtool \
+    	libyaml-dev make pkg-config sqlite3 zlib1g-dev libgmp-dev \
+    	libreadline-dev libssl-dev
+  sudo gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
+  curl -sSL https://get.rvm.io | sudo bash -s stable
+  sudo usermod -a -G rvm `whoami`
+  ```
+
+  If `gpg --keyserver` gives an error, you can use:
+  ```bash
+    sudo gpg --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
+  ```
+  or
+  ```bash
+    sudo gpg2 --keyserver hkp://pool.sks-keyservers.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
+  ```
+  Restart your WSL VM before moving forward.
+
+- For installing Ruby, run:
+  ```bash
+    rvm install ruby
+    rvm --default use ruby 2.7.1
+    gem update
+    gem install jekyll bundler
+  ```
+
+### For Linux
+
+- Prerequisites
+  ```bash
+    sudo apt-get update
+    sudo apt-get install autoconf bison build-essential libssl-dev libyaml-dev libreadline6-dev zlib1g-dev libncurses5-dev libffi-dev libgdbm3 libgdbm-dev
+  ```
+#### Installing rbenv
+
+- Cloning the rbenv repository
+  ```bash
+    git clone https://github.com/rbenv/rbenv.git ~/.rbenv
+  ```
+- Setting the path
+  ```bash
+    echo 'export PATH="$HOME/.rbenv/bin:$PATH"' >> ~/.bashrc
+  ```
+- rbenv init
+  ```bash
+    echo 'eval "$(rbenv init -)"' >> ~/.bashrc
+  ```
+- Reload your bashrc
+  ```bash
+    source ~/.bashrc
+  ```
+  <strong>Note:</strong> Change bashrc with your shell specific rc file, for eg: if you are using zsh then the filename is zshrc.
+  
+- Check installation
+  ```bash
+    type rbenv
+  ```
+#### Install Ruby
+
+- rbenv install version
+```bash
+  rbenv install 2.5.1
+```
+- To list all the versions that can be installed
+```bash
+  rbenv install --list-all
+```
+- Set which Ruby version you want to use
+```bash
+  rbenv global version
+```
+- Check Ruby installation
+```bash
+  ruby -v
+```
 
-Alternatively, if you are running Windows 10, you may install the Windows Subsystem for Linux:
+### For MacOs
 
-- [WSL1](https://docs.microsoft.com/en-us/windows/wsl/install-win10) for Windows build version 1607 or higher.
+- Use docs here [Jekyll installation](https://jekyllrb.com/docs/installation/macos/)
+- After installing Ruby and Jekyll, run the following:
+```bash
+  bundle install
+  bundle config build.puma --with-cflags="-Wno-error=implicit-function-declaration"
+  gem install puma:4.3.5 -- --with-cflags="-Wno-error=implicit-function-declaration"
```

Comment:
Is this necessary? I didn't have to perform this on my Mac...

Reply:
Hey, I'm not a Mac user, could you please help me out with the Mac part, the documentation provided had these instructions so I thought of including those, anyways I will be updating with all the changes!

Reply:
Sure. For MacOS, let's just point contributors to https://jekyllrb.com/docs/.